File: src/de/grobox/liberario/activities/AboutActivity.java
/*    Liberario
 *    Copyright (C) 2013 Torsten Grote
 *
 *    This program is Free Software: you can redistribute it and/or modify
 *    it under the terms of the GNU General Public License as
 *    published by the Free Software Foundation, either version 3 of the
 *    License, or (at your option) any later version.
 *
 *    This program is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU General Public License for more details.
 *
 *    You should have received a copy of the GNU General Public License
 *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package de.grobox.liberario.activities;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import de.grobox.liberario.R;
import de.grobox.liberario.fragments.FragmentAbout;
import de.grobox.liberario.ui.SlidingTabLayout;

public class AboutActivity extends AppCompatActivity {
	AboutPagerAdapter mPagerAdapter;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_about);

		Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
		if(toolbar != null) {
			setSupportActionBar(toolbar);

			ActionBar actionBar = getSupportActionBar();
			if(actionBar != null) actionBar.setDisplayHomeAsUpEnabled(true);
		}

		setTitle(getResources().getString(R.string.action_about) + " " + getResources().getString(R.string.app_name));

		ViewPager viewPager = (ViewPager) findViewById(R.id.pager);

		// don't recreate the fragments when changing tabs
		viewPager.setOffscreenPageLimit(3);

		mPagerAdapter = new AboutPagerAdapter(getSupportFragmentManager());
		viewPager.setAdapter(mPagerAdapter);

		SlidingTabLayout slidingTabLayout = (SlidingTabLayout) findViewById(R.id.sliding_tabs);
		slidingTabLayout.setDistributeEvenly(true);
		slidingTabLayout.setSelectedIndicatorColors(getResources().getColor(R.color.accent));
		slidingTabLayout.setViewPager(viewPager);
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		if(item.getItemId() == android.R.id.home) {
			onBackPressed();
			return true;
		} else {
			return false;
		}
	}


	class AboutPagerAdapter extends FragmentStatePagerAdapter {
		// Since this is an object collection, use a FragmentStatePagerAdapter,
		// and NOT a FragmentPagerAdapter.
		public AboutPagerAdapter(FragmentManager fm) {
			super(fm);
		}

		@Override
		public Fragment getItem(int i) {
			if(i == 1) {
				// TODO: Developers
				return new FragmentAbout();
			} else if(i == 2) {
				// TODO: Libraries
				return new FragmentAbout();
			}
			return new FragmentAbout();
		}

		@Override
		public int getCount() {
			return 3;
		}

		@Override
		public CharSequence getPageTitle(int i) {
			if(i == 1) {
				return getString(R.string.tab_devs);
			} else if(i == 2) {
				return getString(R.string.tab_libraries);
			}
			return getString(R.string.tab_about);
		}
	}
}

File: app/src/main/java/de/grobox/transportr/about/AboutActivity.kt
/*
 *    Transportr
 *
 *    Copyright (c) 2013 - 2021 Torsten Grote
 *
 *    This program is Free Software: you can redistribute it and/or modify
 *    it under the terms of the GNU General Public License as
 *    published by the Free Software Foundation, either version 3 of the
 *    License, or (at your option) any later version.
 *
 *    This program is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU General Public License for more details.
 *
 *    You should have received a copy of the GNU General Public License
 *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package de.grobox.transportr.about

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.mikepenz.aboutlibraries.LibsBuilder
import de.grobox.transportr.R
import de.grobox.transportr.TransportrActivity
import de.grobox.transportr.databinding.ActivityAboutBinding

class AboutActivity : TransportrActivity() {

    companion object {
        @JvmField
        val TAG : String = AboutActivity::class.java.simpleName
    }

    private lateinit var binding: ActivityAboutBinding;

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAboutBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setUpCustomToolbar(false)

        val pager = binding.pager
        pager.adapter = AboutPagerAdapter(supportFragmentManager)

        val tabLayout = binding.tabLayout;
        tabLayout.setupWithViewPager(pager)
    }

    private inner class AboutPagerAdapter internal constructor(fm: FragmentManager) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

        override fun getItem(i: Int): Fragment {
            return when (i) {
                0 -> AboutFragment()
                1 -> LibsBuilder()
                    // Pass the fields of your application to the lib so it can find all external lib information
                    .withFields(R.string::class.java.fields)
                    // get the fragment
                    .supportFragment()
                else -> throw IllegalArgumentException()
            }
        }

        override fun getCount(): Int {
            return 2
        }

        override fun getPageTitle(i: Int): CharSequence? {
            return when (i) {
                0 -> getString(R.string.tab_about)
                1 -> getString(R.string.tab_libraries)
                else -> throw IllegalArgumentException()
            }
        }

    }

}
